File: daily/406.根据身高重建队列.js
/*
 * @lc app=leetcode.cn id=406 lang=javascript
 *
 * [406] 根据身高重建队列
 */

// @lc code=start
/**
 * @param {number[][]} people
 * @return {number[][]}
 */
var reconstructQueue = function(people) {
  people.sort((a, b) => a[0] === b[0] ? a[1] - b[1] : b[0] - a[0])
  return people.reduce((acc, cur) => (acc.splice(cur[1], 0, cur), acc), [])
};
// @lc code=end

const people = [[7,0],[4,4],[7,1],[5,0],[6,1],[5,2]]
const result = reconstructQueue(people)
console.log(result)

File: daily/406.根据身高重建队列.rs
/*
 * @lc app=leetcode.cn id=406 lang=rust
 *
 * [406] 根据身高重建队列
 */

// @lc code=start
impl Solution {
    pub fn reconstruct_queue(people: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let mut people = people;
        let mut result = vec![];
        people.sort_by(|a, b| if a[0] == b[0] { a[1].cmp(&b[1]) } else { b[0].cmp(&a[0]) } );
        for p in people { result.insert(p[1] as usize, p) }
        return result;
    }
}
// @lc code=end

